prow/cmd/deck/static/api/prow.ts

```diff
@@ -1,15 +1,38 @@
 export type JobType = "presubmit" | "postsubmit" | "batch" | "periodic";
 export type JobState = "triggered" | "pending" | "success" | "failure" | "aborted" | "error" | "unknown" | "";
 
-export interface Job {
-  type: JobType;
+// Pull describes a pull request at a particular point in time.
+// Pull mirrors the Pull struct defined in types.go.
+export interface Pull {
+  number: number;
```

Comment:
Every field here is marked as `omitempty` in types.go, and so should be marked as optional here.

```ts
export interface Pull {
  number?: number;
  author:? string;
  sha?: string;
  title?: string;
  ref?: string;
  link?: string;
  commit_link?: string;
  author_link?: string;
}
```

Reply:
Addressed this, but I wasn't sure about the best way to handle missing or empty values. It wasn't clear to me whether the existing code handles empty strings correctly (or even how to define correctness here), but wherever possible I tried not to change the existing behavior (usually by doing `|| ""`) (except in the few places I was fairly sure it wasn't desirable). PTAL

Reply:
The alternative would be to drop the `omitempty` qualifiers on the Go side if it doesn't actually make sense for properties to be omitted.

Reply:
Having looked through the code, there are a significant number of assumptions that these will be non-empty and contain something useful. I would consider whether it really makes sense for these fields to ever be omitted (I suspect it does not), and if not remove the `omitempty` declarations and also remove the handling for the possibility of them being omitted.

Reply:
Okay, I removed `omitempty` from the following fields: `Refs.Repo`, `Refs.Org`, `Pull.Number`, `Pull.Author`, `Pull.SHA`. Seems reasonable as those structs would be pretty useless without these fields.